Assets/Shopify/UIToolkit/Theme Controllers/Editor/SingleProductThemeControllerEditor.cs

```diff
@@ -3,54 +3,51 @@
     using UnityEditor;
     using System;
 
-    [CustomEditor(typeof(SingleProductThemeController))]
-    public class SingleProductThemeControllerEditor : Editor {
-        public ISingleProductThemeControllerEditorView View;
-        public IShopCredentialsView CredentialsView;
+    public interface ISingleProductThemeControllerEditorView : IBaseThemeControllerEditorView {
+        void DrawProductPicker();
+    }
 
-        private ShopCredentialsVerifier _verifier {
+    [CustomEditor(typeof(SingleProductThemeController))]
+    public class SingleProductThemeControllerEditor : BaseThemeControllerEditor, ISingleProductThemeControllerEditorView {
+        public new SingleProductThemeController Target {
             get {
-                if (_cachedVerifier == null) {
-                    _cachedVerifier = new ShopCredentialsVerifier(_credentials);
-                }
-                return _cachedVerifier;
+                return (SingleProductThemeController) target;
             }
         }
 
-        private ShopCredentialsVerifier _cachedVerifier;
-
-        private IShopCredentials _credentials {
+        public new ISingleProductThemeControllerEditorView View {
             get {
-                return target as IShopCredentials;
+                return (ISingleProductThemeControllerEditorView) base.View;
             }
-        }
 
-        public SingleProductThemeController Target {
-            get {
-                return target as SingleProductThemeController;
+            set {
+                base.View = value;
             }
         }
 
-        public void OnEnable() {
-            if (Target == null) return;
+        public override void OnEnable() {
+            base.OnEnable();
+            View = View ?? this;
+        }
 
-            View = new SingleProductThemeControllerEditorView();
-            CredentialsView = new ShopCredentialsView(_verifier);
-            BindThemeIfPresent();
+        private ProductPicker _productPicker {
+            get {
+                _cachedPicker = _cachedPicker ?? new ProductPicker(Client);
+                return _cachedPicker;
+            }
         }
+        private ProductPicker _cachedPicker;
 
-        public override void OnInspectorGUI() {
-            CredentialsView.DrawInspectorGUI(serializedObject);
+        public override void OnShowConfiguration() {
+            View.DrawProductPicker();
+        }
 
-            if (Target.Theme == null) {
-                View.ShowThemeHelp();
-                return;
-            }
+        protected override void OnClientChanged() {
+            _productPicker.Client = Client;
         }
 
-        private void BindThemeIfPresent() {
-            if (Target.Theme != null) return;
-            Target.Theme = Target.GetComponent<ISingleProductTheme>();
+        void ISingleProductThemeControllerEditorView.DrawProductPicker() {
```

Comment:
Why do we use `ISingleProductThemeControllerEditorView.DrawProductPicker` here.

Reply:
its called explicit interface implementation and it means that this method will only be available when the class is cast to `ISingleProductThemeControllerEditorView`, its just a safeguard that keeps us from using the view methods without casting the class to a view, which means that refactoring the view in the future will be easier (such as extracting it to a new class or performing dependency injection, etc).

Reply:
Yeah I was just mainly wondering about why you wanted to do this.

Reply:
the view concept in general? This is primarily so that we can narrow the untested surface of the controllers to only include the actual methods that touch the editor GUI (because we don't know how to test the editor). It allows us to mock the view dependency away and assert that particular things are being "told to draw" given certain states in the editor.

Reply:
Nah I was wondering mainly about explicitly defining the interface implementation since there are no other interfaces associated with this class.